src/adapter/src/coord.rs

```diff
@@ -2347,10 +2349,17 @@ impl Coordinator {
         // Add builtin table updates the clear the contents of all system tables
         debug!("coordinator init: resetting system tables");
         let read_ts = self.get_local_read_ts().await;
-        for system_table in entries
-            .iter()
-            .filter(|entry| entry.is_table() && entry.id().is_system())
-        {
+        let mz_storage_usage_by_shard_schema: SchemaSpecifier = self
+            .catalog()
+            .resolve_system_schema(MZ_STORAGE_USAGE_BY_SHARD.schema)
+            .into();
+        let is_storage_usage_by_shard = |entry: &CatalogEntry| -> bool {
+            entry.name.item == MZ_STORAGE_USAGE_BY_SHARD.name
+                && entry.name.qualifiers.schema_spec == mz_storage_usage_by_shard_schema
+        };
+        for system_table in entries.iter().filter(|entry| {
+            entry.is_table() && entry.id().is_system() && !is_storage_usage_by_shard(entry)
```

Comment:
Is it possible to panic/return an error if the fingerprint used to determine builtin migrations changes for the new `mz_storage_usage_by_shard` table?

Reply:
That was added here: https://github.com/MaterializeInc/materialize/blob/e1725417265f7295417bac728a51f75ff9c91f33/src/adapter/src/catalog/open.rs#L951

Unless I'm misunderstanding you.

Reply:
Ahhh sorry I missed this, looks great!